tasks/sep/fp-recovery/005-set-game-implementation/justfile

```diff
@@ -94,3 +96,107 @@ set-implementation gameType='' implAddr='':
   INFO="* Set implementation for game type {{gameType}} to {{implAddr}} in \`DisputeGameFactory\` ${DGF_ADDR}: \`setImplementation({{gameType}}, {{implAddr}})\`"
   sed -e "s|<!--NEXT TASK DESCRIPTION-->|${INFO}\n<!--NEXT TASK DESCRIPTION-->|" out/README.md > out/README.md.tmp
   mv out/README.md.tmp out/README.md
+
+copy-anchor-state fromGameType='' toGameTypes='':
+  #!/usr/bin/env bash
+  set -euo pipefail
+  . out/.env
+  export ETH_RPC_URL # Make available to cast since the .env file is in the out subdir.
+
+  DGF_ADDR=$(cast call "${SYSTEM_CONFIG}" "disputeGameFactory()(address)")
+
+  # Find the anchor state registry for the source game.
+  SOURCE_IMPL=$(cast call "${DGF_ADDR}" 'gameImpls(uint32)(address)' '{{fromGameType}}')
+  REGISTRY_PROXY=$(cast call "${SOURCE_IMPL}" 'anchorStateRegistry()(address)' "${SOURCE_IMPL}")
+  REGISTRY_IMPL=$(cast call "${REGISTRY_PROXY}" 'implementation()(address)')
+  SUPERCHAIN_CONFIG=$(cast call "${REGISTRY_PROXY}" 'superchainConfig()(address)')
+
+  # Load the current anchor state
+  ANCHOR_JSON=$(cast call -j "${REGISTRY_PROXY}" 'anchors(uint32)(bytes32,uint256)' '{{fromGameType}}')
```

Comment:
Looks like this `--json` arg is only on foundry versions newer then what we specify in [`versions.json`](https://github.com/ethereum-optimism/optimism/blob/94ab846ab439ef26802ca32d6bc5921aadbafa0f/versions.json#L4)—I had to upgrade from the monorepo version to latest for this to work

Reply:
hmm, `cast install-foundry` in monorepo gives me `cast 0.2.0 (143abd6 2024-09-04T00:24:42.542256000Z)` which does have the `-j` option for call and works for me.

Reply:
ah maybe I was just on an even older version then (I've already upgraded, so can't confirm)